cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/upload/TextStreamBodyConsumer.java

```diff
@@ -106,23 +105,30 @@ public void handleError(Throwable cause) {
     LOG.warn("Failed to handle upload to stream {}", streamId, cause);
   }
 
-  private void processChunk(final ChannelBuffer chunk) throws IOException {
+  private void processChunk(final ByteBuf chunk) throws IOException {
     final ContentWriter writer = getContentWriter();
     writer.appendAll(new AbstractIterator<ByteBuffer>() {
       @Override
       protected ByteBuffer computeNext() {
-        int len = chunk.bytesBefore(ChannelBufferIndexFinder.CRLF);
-        if (len <= 0) {
+        int len = chunk.bytesBefore((byte) '\n');
+        if (len < 0) {
           return endOfData();
         }
-        ChannelBuffer body = chunk.readSlice(len);
-        // Skip the CRLF.
-        while (chunk.bytesBefore(ChannelBufferIndexFinder.CRLF) == 0) {
-          chunk.readByte();
+        ByteBuf body;
+        if (len > 0) {
+          body = chunk.readSlice(len);
```

Comment:
seems like this is duplicate code from elsewhere in this PR.
Can't refactor it?

Reply:
I meant the next few lines

Reply:
They are doing different logic.